src/test/kotlin/com/xtyrrell/maxonbank/command/AccountTests.kt

```diff
@@ -0,0 +1,116 @@
+package com.xtyrrell.maxonbank.command
+
+import com.xtyrrell.maxonbank.coreapi.*
+import org.axonframework.test.aggregate.AggregateTestFixture
+import org.axonframework.test.aggregate.FixtureConfiguration
+import org.axonframework.test.matchers.Matchers
+import org.junit.jupiter.api.*
+import java.util.*
+
+
+class AccountTests {
+    private lateinit var fixture: FixtureConfiguration<Account>
+
+    @BeforeEach
+    fun setUp() {
+        fixture = AggregateTestFixture(Account::class.java)
+    }
+
+    @Nested
+    @DisplayName("OpenAccountCommand")
+    @TestInstance(TestInstance.Lifecycle.PER_CLASS)
+    inner class OpenAccountCommandTests {
+        @Test
+        fun `should open an account`() {
+            fixture.givenNoPriorActivity()
+                .`when`(OpenAccountCommand())
+                .expectSuccessfulHandlerExecution()
+                .expectEventsMatching(
+                    Matchers.exactSequenceOf(
+                        Matchers.messageWithPayload(
+                            Matchers.matches { payload: AccountOpenedEvent ->
+                                // This is weird, but I think it works. If we can cast [payload] to an
+                                // AccountOpenedEvent, then it must be one. If [payload] is any other type, I believe
+                                // this will throw an exception.
+                                // Still, it would be nicer to rather do this check at runtime, something like
+                                // `return payload is AccountOpenedEvent`
+                                true
+                            }
+                        )
+                    )
+                )
+        }
+    }
+
+    @Nested
+    @DisplayName("DepositFundsCommand")
+    @TestInstance(TestInstance.Lifecycle.PER_CLASS)
+    inner class DepositFundsCommandTests {
+        @Test
+        fun `should allow depositing funds for a non-zero positive amount`() {
+            val accountId = UUID.randomUUID()
+
+            fixture
+                .given(AccountOpenedEvent(accountId))
+                .`when`(DepositFundsCommand(accountId, 500))
+                .expectSuccessfulHandlerExecution()
+                .expectEvents(FundsDepositedEvent(accountId, 500))
+        }
+
+        @Test
+        fun `should prevent depositing funds for zero or negative amount`() {
```

Comment:
I would split these two tests, and always only have one call the fixture per test. This makes it much easier to debug what broke when one of the tests starts failing (and ensures your `BeforeEach` is run before each call to fixture)